Lib/argparse.py

```diff
@@ -1360,7 +1360,7 @@ def __init__(self,
         self._defaults = {}
 
         # determines whether an "option" looks like a negative number
-        self._negative_number_matcher = _re.compile(r'^-\d+$|^-\d*\.\d+$')
+        self._negative_number_matcher = _re.compile(r'^-?\d[\d_]*(\.\d+)?$')
```

Comment:
Just curious, why is this `?` here? it seems like this new regex will now match positive numbers as well?

```suggestion
        self._negative_number_matcher = _re.compile(r'^-\d[\d_]*(\.\d+)?$')
```

Reply:
Ooooops, good catch! I was playing around with this in regex101.com and forgot to drop that once I was happy with the expression.